### PR TITLE
logging intro

### LINKED ISSUE #1282: Write introduction on logging
When we switched to `tracing` I left a [TODO](https://github.com/Smithay/smithay/blob/164855cf4bb7c95a677f829a3392bd39a312d80e/src/lib.rs#L49).
Before creating a release we should fill that TODO.

### PR DESCRIPTION
**Pull request by cmeissl**
~depends on #1283~ merged

fixes #1282

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -48,7 +48,17 @@
 //!
 //! ### Logging
 //!
-//! TODO: docs about tracing
+//! Smithay makes extensive use of [`tracing`] for it's internal logging.
```
Comment (YaLTeR) [inline src/lib.rs:51]:
```suggestion
//! Smithay makes extensive use of [`tracing`] for its internal logging.
```

Comment (cmeissl) [reply src/lib.rs]: done, thanks!

**Review by YaLTeR — APPROVED** [review]

**Comment by Drakulix** [comment]
Could we maybe add something like:

If you do not want to use tracing for your compositor, refer to [`log` compatibility](https://docs.rs/tracing/latest/tracing/#log-compatibility) for how to forward smithays debug output to other `log` compatible frameworks.

**Comment by cmeissl** [comment]
> Could we maybe add something like:
> 
> If you do not want to use tracing for your compositor, refer to [`log` compatibility](https://docs.rs/tracing/latest/tracing/#log-compatibility) for how to forward smithays debug output to other `log` compatible frameworks.

Sure, this is definitely a good idea, added.

**Review by Drakulix — APPROVED** [review]

**Pull request merged**